services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java

```diff
@@ -0,0 +1,104 @@
+package app.coronawarn.server.services.federation.upload.runner;
+
+import app.coronawarn.server.common.persistence.domain.DiagnosisKey;
+import app.coronawarn.server.common.persistence.repository.TestDataUploadRepository;
+import app.coronawarn.server.common.protocols.external.exposurenotification.ReportType;
+import app.coronawarn.server.services.federation.upload.config.UploadServiceConfig;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(-1)
+@Profile("testdata")
+public class TestDataGeneration implements ApplicationRunner {
+
+  private final UploadServiceConfig uploadServiceConfig;
+  private final Logger logger = LoggerFactory.getLogger(TestDataGeneration.class);
+  private final TestDataUploadRepository keyRepository;
+
+  public TestDataGeneration(UploadServiceConfig uploadServiceConfig,
+      TestDataUploadRepository keyRepository) {
+    this.uploadServiceConfig = uploadServiceConfig;
+    this.keyRepository = keyRepository;
+  }
+
+  private static byte[] randomByteData() {
+    byte[] keydata = new byte[16];
+    new Random().nextBytes(keydata);
```

Comment:
consider usage of: `java.util.concurrent.ThreadLocalRandom` instead of 'normal' random